synapse/handlers/e2e_keys.py

```diff
@@ -496,6 +496,20 @@ async def upload_keys_for_user(self, user_id, device_id, keys):
             log_kv(
                 {"message": "Did not update one_time_keys", "reason": "no keys given"}
             )
+        fallback_keys = keys.get("fallback_keys", None)
+        if fallback_keys and isinstance(fallback_keys, dict):
```

Comment:
If this is not a dict should we give a better error message than "no keys given"?